sio.go

```diff
@@ -115,6 +115,42 @@ type Config struct {
 	PayloadSize int
 }
 
+// EncryptedSize returns the number of bytes of an DARE 2.0
+// encrypted data stream where size is the size of the plaintext
+// in bytes. It returns an error if size is larger than 2^48.
+func EncryptedSize(size uint64) (uint64, error) {
+	if size > 1<<48 {
+		return 0, errors.New("sio: size is too large for DARE")
+	}
+	encSize := (size / maxPayloadSize) * maxPackageSize
+	if mod := size % maxPayloadSize; mod > 0 {
+		encSize += mod + (headerSize + tagSize)
+	}
+	return encSize, nil
+}
+
+// DecryptedSize returns the number of bytes of an decrypted data
+// stream where size is the size of an DARE 2.0 encrypted data in
+// bytes. It returns an error if size is larger than 2^48 + 2^37 or
+// size is not a valid size for an encrypted data stream.
+func DecryptedSize(size uint64) (uint64, error) {
+	if size == 0 {
+		return 0, nil
+	}
+	const maxStreamSize = (1 << 48) + (32 * (1 << 32))
+	if size > maxStreamSize {
+		return 0, errors.New("sio: size is too large for DARE")
+	}
+	decSize := (size / maxPackageSize) * maxPayloadSize
+	if mod := size % maxPackageSize; mod > 0 {
+		if mod <= headerSize+tagSize {
+			return 0, errors.New("sio: size is not valid") // last package is not valid
```

Comment:
on server we use something like errObjectTampered() can we do that here as well?

Reply:
On the server we can simply do:
```
if size, err = sio.DecryptSize(size); err != nil {
     return errObjectTampered
}
```
If we want to export an error type we actually have to use it also for decrypting operations too (like decrypt reader/writer). We can export eg. `sio.AuthError` but we may should address separately.

Reply:
See #24